components/nexus-core/src/main/java/org/sonatype/nexus/internal/blobstore/BlobStoreManagerImpl.java

```diff
@@ -81,39 +119,102 @@ protected void doStop() throws Exception {
 
   @Override
   @Guarded(by = STARTED)
+  public Iterable<BlobStore> browse() {
+    return ImmutableList.copyOf(stores.values());
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public BlobStore create(final BlobStoreConfiguration configuration) throws Exception {
+    checkNotNull(configuration);
+    log.debug("Creating blob-store: {} with content root: {}", configuration.getName(),
+        configuration.getPath());
+
+    store.create(configuration);
+
+    BlobStore blobStore = newBlobStore(configuration);
+    track(configuration.getName(), blobStore);
+    
+    blobStore.start();
+    //TODO - event publishing
+
+    return blobStore;
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public BlobStore update(final BlobStoreConfiguration configuration) throws Exception {
+    checkNotNull(configuration);
+    String name = checkNotNull(configuration.getName());
+    String path = checkNotNull(configuration.getPath());
+    
+    log.debug("Updating BlobStore: {}, {}", name, path);
+    
+    //TODO - changing the path will leave behind previous content, but is that sensible?
+    BlobStore blobStore = blobStore(name);
+    store.update(configuration);
+    blobStore.stop();
+    blobStore = newBlobStore(configuration);
+    blobStore.start();
+    track(configuration.getName(), blobStore);
+    return blobStore;
+  }
+
+  @Override
+  @Guarded(by = STARTED)
+  public void delete(final String name) throws Exception {
+    checkNotNull(name);
+    
+    log.debug("Deleting BlobStore: {}", name);
+    BlobStore blobStore = blobStore(name);
+    blobStore.stop();
+    Iterable<BlobStoreConfiguration> filtered = Iterables.filter(store.list(), new Predicate<BlobStoreConfiguration>()
+    {
+      @Override
+      public boolean apply(final BlobStoreConfiguration input) {
+        return input.getName().equals(name);
+      }
+    });
+    if(!filtered.iterator().hasNext()) {
+      throw new IllegalStateException("Could not find configuration to delete: " + name);
+    }
+    store.delete(filtered.iterator().next());
+    untrack(name);
+    
+    //TODO - event publishing
+  }
+
+  @Override
+  @Guarded(by = STARTED)
   public BlobStore get(final String name) {
     checkNotNull(name);
 
     synchronized (stores) {
       BlobStore store = stores.get(name);
 
+      // TODO - remove auto-create functionality?
```

Comment:
I left this in place to support existing usage, but likely we want to ditch in favor of explicit creation by consumers